src/CloudGraphsDFG/services/CloudGraphsDFG.jl

```diff
@@ -708,3 +708,126 @@ function setSolvable!(dfg::CloudGraphsDFG, sym::Symbol, solvable::Int)::Int
+
+end
+
+function removeTags!(dfg::CloudGraphsDFG, sym::Symbol, tags::Vector{Symbol})
+
+    if isVariable(dfg,sym)
+        getNode = getVariable
+        updateNode! = updateVariable!
+    else
+        getNode = getFactor
+        updateNode! = updateFactor!
+    end
+
+    node = getNode(dfg, sym)
+    setdiff!(getTags(node), tags)
+
+    updateNode!(dfg, node)
+
+    return getTags(getNode(dfg, sym))
+
+end
+
+function emptyTags!(dfg::CloudGraphsDFG, sym::Symbol)
+
+    if isVariable(dfg,sym)
+        getNode = getVariable
+        updateNode! = updateVariable!
+    else
+        getNode = getFactor
+        updateNode! = updateFactor!
+    end
+
+    node = getNode(dfg, sym)
+    empty!(getTags(node))
+
+    updateNode!(dfg, node)
+
+    return getTags(getNode(dfg, sym))
+
+end
+
+
+
+function RESERVED_mergeTags!(dfg::CloudGraphsDFG, sym::Symbol, tags::Vector{Symbol})
+
+    nodeId = _tryGetNeoNodeIdFromNodeLabel(dfg.neo4jInstance,
+                                            dfg.userId,
+                                            dfg.robotId,
+                                            dfg.sessionId,
+                                            sym)
+
+    neo4jNode = Neo4j.getnode(dfg.neo4jInstance.graph, nodeId)
+
+    addnodelabels(neo4jNode, string.(tags))
+
+    return Set(setdiff(Symbol.(getnodelabels(neo4jNode)), Symbol.([dfg.userId, dfg.robotId, dfg.sessionId]), [sym]))
+
+end
+
+
+function RESERVED_removeTags!(dfg::CloudGraphsDFG, sym::Symbol, tags::Vector{Symbol})
+
+    nodeId = _tryGetNeoNodeIdFromNodeLabel(dfg.neo4jInstance,
+                                            dfg.userId,
+                                            dfg.robotId,
+                                            dfg.sessionId,
+                                            sym)
+
+    neo4jNode = Neo4j.getnode(dfg.neo4jInstance.graph, nodeId)
+
+    shouldStay = Symbol.([dfg.userId, dfg.robotId, dfg.sessionId]) ∪ [sym, :VARIABLE, :FACTOR]
+
+    for tag in tags
+        if tag in shouldStay
+            @warn("Label:$tag is not allowed to be removed from tags and will be ignored.")
+        else
+            deletenodelabel(neo4jNode, string(tag))
+        end
+    end
+
+    return Set(setdiff(Symbol.(getnodelabels(neo4jNode)), shouldStay))
+
+end
+
+
+function RESERVED_emptyTags!(dfg::CloudGraphsDFG, sym::Symbol)
+
+    nodeId = _tryGetNeoNodeIdFromNodeLabel(dfg.neo4jInstance,
+                                            dfg.userId,
+                                            dfg.robotId,
+                                            dfg.sessionId,
+                                            sym)
+
+    neo4jNode = Neo4j.getnode(dfg.neo4jInstance.graph, nodeId)
+
+    shouldStay = Symbol.([dfg.userId, dfg.robotId, dfg.sessionId]) ∪ [sym, :VARIABLE, :FACTOR]
+    tags = setdiff(Symbol.(getnodelabels(neo4jNode)), shouldStay)
+    # tags = Symbol.(getnodelabels(neo4jNode))
+
+    return removeTags!(dfg, sym, tags)
+end
```

Comment:
@GearsAD If you have a few minutes, take a look at this. I'm thinking of not having the Neo4j labels repeated as tags in the variable data. But I implemented the variable tags version.

Reply:
That's fair actually, we can subtract them, to leave the important ones like LANDMARK etc. I don't think there's a need for the user/robot/session labels. As long as they're on the neo4j node we're ok, that helps us search more efficiently, but the user doesn't need to see that. If I understand correctly, is that what you mean?